libraries/AP_Stats/AP_Stats.cpp

```diff
@@ -122,3 +131,24 @@ void AP_Stats::set_flying(const bool is_flying)
         _flying_ms = 0;
     }
 }
+
+/*
+  get flight distance since boot
+ */
+uint32_t AP_Stats::get_flight_distance_m(void)
+{
+    update_flighttime();
+    AP_AHRS &ahrs = AP::ahrs();
+    Vector2f v = ahrs.groundspeed_vector();
+    float speed = v.length();
+    uint64_t time = _flying_ms - _last_distance_ms;
+    _last_distance_ms = _flying_ms;
+    uint32_t delta = (time * speed)/1000; //meters
```

Comment:
okay, will setup fltdistance,and delta as floats to maintain the precision, correct?been a while since programming for real, knew I was casting, but thought the cast happened after the calculation...duh